### ISSUE TITLE
Failed to build gcc 8/10/11 for 64bit: "error: implicit declaration of function 'getpagesize'" (missing mmap)

### ISSUE THREAD

**Issue by kmatheussen** [mxe/mxe#2813]
```
/home/kjetil/mxe/tmp-gcc-x86_64-w64-mingw32.shared/gcc-11.2.0/libbacktrace/mmapio.c: In function 'backtrace_get_view':
/home/kjetil/mxe/tmp-gcc-x86_64-w64-mingw32.shared/gcc-11.2.0/libbacktrace/mmapio.c:73:14: error: implicit declaration of function 'getpagesize' [-Werror=implicit-function-declaration]
   73 |   pagesize = getpagesize ();
      |              ^~~~~~~~~~~
cc1: all warnings being treated as errors
Makefile:1401: recipe for target 'mmapio.lo' failed
```

This patch might fix it: https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=47f4703c33c4936fd423c2a1180b2de144115d3d
I tried to apply it, but it didn't apply cleanly, so I'm not sure if it'll work.

[gcc_x86_64-w64-mingw32.shared.gz](https://github.com/mxe/mxe/files/8464920/gcc_x86_64-w64-mingw32.shared.gz)

**Comment by kmatheussen** [comment]
Same error when trying to compile gcc10. It's not happening for i686.

**Comment by kmatheussen** [comment]
Same error with gcc8 as well. That's strange because I've compiled gcc8 successfully before.

**Comment by tonytheodore** [comment]
Are you rebuilding gcc after mman-win32 is installed? These might be similar issues #2526 #2529.

I think you can try:
```
make gcc gcc_CONFIGURE_OPTS=ac_cv_header_sys_mman_h=no
```
or adding a line to remove `sys/mman.h` here https://github.com/mxe/mxe/blob/master/src/gcc.mk#L66.

**Comment by kmatheussen** [comment]
Yes, mman-win32 seems to have been installed after gcc was originally
installed:

***@***.***:~/mxe$ ls -la
./log/20201222_170347/mman-win32_x86_64-w64-mingw32.shared
-rw-rw-r-- 1 kjetil kjetil 36999 des.  22  2020
./log/20201222_170347/mman-win32_x86_64-w64-mingw32.shared

I haven't installed it manually so I guess it must have been pulled in as a
dependency for something.

Now I was upgrading the qt5 package (after doing "git pull"), and then it
also wanted to upgrade, or perhaps recompile, gcc.
